apps/spruce/src/pages/image/PackagesTable/PackagesTable.test.tsx

```diff
@@ -0,0 +1,318 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { RenderFakeToastContext } from "context/toast/__mocks__";
+import {
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_PACKAGES } from "gql/queries";
+import {
+  renderWithRouterMatch as render,
+  screen,
+  waitFor,
+  within,
+  userEvent,
+} from "test_utils";
+import { ApolloMock } from "types/gql";
+import { PackagesTable } from ".";
+
+const wrapper = ({ children }: { children: React.ReactNode }) => (
+  <MockedProvider
+    mocks={[
+      imagePackagesPageOneMock,
+      imagePackagesNameFilterMock,
+      imagePackagesPageTwoMock,
+    ]}
+  >
+    {children}
+  </MockedProvider>
+);
+
+describe("packages table", () => {
+  it("shows name field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <PackagesTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("packages-table-row")).toHaveLength(10);
+    });
+    const expectedNames: string[] = [];
+    for (let i = 0; i < 10; i++) {
+      const packageData =
+        imagePackagesPageOneMock.result?.data?.image?.packages.data[i];
+      if (packageData && packageData.name) {
+        expectedNames.push(packageData.name);
+      }
+    }
+    const rows = screen.getAllByDataCy("packages-table-row");
+    for (let i = 0; i < expectedNames.length; i++) {
+      expect(within(rows[i]).getAllByRole("cell")[0]).toHaveTextContent(
+        expectedNames[i],
+      );
+    }
+  });
+
+  it("shows manager field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <PackagesTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("packages-table-row")).toHaveLength(10);
+    });
+    const rows = screen.getAllByDataCy("packages-table-row");
+    for (let i = 0; i < 10; i++) {
+      expect(within(rows[i]).getAllByRole("cell")[1]).toHaveTextContent(
+        "pip 22.0.2 from (python 3.10)",
+      );
+    }
+  });
+
+  it("shows version field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <PackagesTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("packages-table-row")).toHaveLength(10);
+    });
+    const expectedVersions: string[] = [];
+    for (let i = 0; i < 10; i++) {
+      const packageData =
+        imagePackagesPageOneMock.result?.data?.image?.packages.data[i];
+      if (packageData && packageData.version) {
+        expectedVersions.push(packageData.version);
+      }
+    }
```

Comment:
I think you could make this look a little cleaner by using the [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) function.